### PR TITLE
Create custom admin article menus

### PR DESCRIPTION
**Pull request by brianteeman**
See #20233 for detailed explanation of the use case

### Test Instructions

Create and enable an admin menu with the "Joomla preset"

Create a new admin menu item of type article list and you can see you now have a full set of filters that you can apply.

For example you can now create menu items in the admin to each category.

If approved I will create the same option for other components

This does not resolve everything on #20233 as it still needs field categories to be implemented for that

Thanks to @tonypartridge for his help

### CODE REVIEW

**Comment by tonypartridge** [comment]
love it, will give it a test soon 👌

File: administrator/components/com_content/views/articles/tmpl/default.xml [hunk 1]
```diff
@@ -5,4 +5,66 @@
...
+				label="COM_MENUS_ADMIN_TAGS_LABEL"
+				description="COM_MENUS_ADMIN_TAGS_DESC"
+				multiple="true"
+			/>
+	
+			<field
+				name="filter_language"
+				type="contentlanguage"
+				label="COM_MENUS_ADMIN_LANGUAGE_LABEL"
+				description="COM_MENUS_ADMIN_LANGUAGE_LABEL"
```
Comment (Quy) [inline administrator/components/com_content/views/articles/tmpl/default.xml]: Change to `COM_MENUS_ADMIN_LANGUAGE_DESC`

File: administrator/language/en-GB/en-GB.com_menus.ini [hunk 2]
```diff
@@ -9,6 +9,18 @@ COM_MENUS_ACTION_DESELECT="Deselect"
 COM_MENUS_ACTION_EXPAND="Expand"
 COM_MENUS_ACTION_SELECT="Select"
 COM_MENUS_ADD_MENU_MODULE="Add a module for this menu"
+COM_MENUS_ADMIN_AUTHOR_DESC="Filter by author."
+COM_MENUS_ADMIN_AUTHOR_LABEL="Author"
+COM_MENUS_ADMIN_CATEGORY_DESC="Filter by Category"
```
Comment (Quy) [inline administrator/language/en-GB/en-GB.com_menus.ini]: Lowercase `C` and add a period.

**Comment by carlitorweb** [comment]
Can add `filters` to the new fields, or in this case is not necessary? following what doing in others PRs

**Comment by brianteeman** [comment]
Sorry I don't understand

File: administrator/components/com_content/views/articles/tmpl/default.xml [hunk 3]
```diff
@@ -5,4 +5,66 @@
...
+				label="COM_MENUS_ADMIN_AUTHOR_LABEL"
+				description="COM_MENUS_ADMIN_AUTHOR_DESC"
+				multiple="true"
+				class="multipleAuthors"
+				>
+				<option value="0">JNONE</option>
+			</field>			
+
+			<field
+				name="filter_tag"
```
Comment (carlitorweb) [inline administrator/components/com_content/views/articles/tmpl/default.xml]: for example, add to this field the `filter="int_array"`

**Comment by tonypartridge** [comment]
No, all this is doing is pre-setting the filters that already exit on a page.

**Comment by mbabker** [comment]
He's talking about input filters (essentially `JFactory::getApplication()->input->get('foo', 'default', $filter);`, not the UI filters.

**Comment by tonypartridge** [comment]
Then of course it is possible as we are just passing the request variables. But in doing that, you have no control over the filters on that page if you pass them in and they work. Note, I haven’t tried filtering the admin ui with filters not existent on the UI. But wouldn’t really advise it.

**Comment by mbabker** [comment]
You're still missing the point here I'm afraid.

The `filter="whatever"` attribute sets what data type something is filtered/validated as when it is read from the request in the JForm API when filtering/validating data.  So if you set `filter="int"` on a field, then when the field's data is read from the request it is filtered/validated as an integer.

**Comment by tonypartridge** [comment]
Ah, clearly I missed that. I just saw the filter on custom fields.

**Comment by carlitorweb** [comment]
@tonypartridge  Sorry for no explain well when i ask about this. Thank you @mbabker

**Comment by brianteeman** [comment]
Gotya. Will update.

**Comment by brianteeman** [comment]
@SharkyKZ can you check the filters please

**Comment by carlitorweb** [comment]
> When you create an admin menu item to display a list of all articles there is no option to select a filter so the article list displays all the articles.

So, this is well covered in this PR. Except, not for the Statue and Access levels. Also, will be good do the same with the featured articles view

**Comment by brianteeman** [comment]
I did think about status and access but there is no point. A users ACL addresses the access and I don't want to pre-filter on status - thats not a good idea.

As for the featured view thats a special type of thing and it doesnt match the use case for this - my 2c

**Comment by carlitorweb** [comment]
Roger that. So, this work perfect!

**Comment by carlitorweb** [comment]
I have tested this item :white_check_mark: successfully on 921f0e1b761ca241ffd220955348c6da725cf7ed<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/20890">issues.joomla.org/tracker/joomla-cms/20890</a>.</sub>

**Comment by SharkyKZ** [comment]
Filters OK.

**Comment by tonypartridge** [comment]
I actually think the access level could be handy for when filtering lots of articles assigned to certain access levels, i.e. ‘Guest Articles’, ‘Members Articles’. ?

Just a thought

**Comment by brianteeman** [comment]
@tonypartridge ah - I was thinking of ACL and not access levels - i will add it

**Comment by carlitorweb** [comment]
I have tested this item :white_check_mark: successfully on 8284fd060da97c02157df88bbb981cb891f19c8d<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/20890">issues.joomla.org/tracker/joomla-cms/20890</a>.</sub>

File: administrator/components/com_content/views/articles/tmpl/default.xml [hunk 4]
```diff
@@ -5,4 +5,79 @@
...
+				name="filter_author_id"
+				type="author"
+				label="COM_MENUS_ADMIN_AUTHOR_LABEL"
+				description="COM_MENUS_ADMIN_AUTHOR_DESC"
+				multiple="true"
+				class="multipleAuthors"
+				filter="int_array"
+				>
+				<option value="0">JNONE</option>
+			</field>
```
Comment (Quy) [inline administrator/components/com_content/views/articles/tmpl/default.xml]: Remove trailing tabs

File: administrator/components/com_content/views/articles/tmpl/default.xml [hunk 5]
```diff
@@ -5,4 +5,79 @@
...
+			/>
+
+			<field
+				name="filter_access"
+				type="accesslevel"
+				label="COM_MENUS_ADMIN_ACCESS_LABEL"
+				description="COM_MENUS_ADMIN_ACCESS_DESC"
+				multiple="true"
+				filter="int_array"
+			/>
```
Comment (Quy) [inline administrator/components/com_content/views/articles/tmpl/default.xml]: Remove trailing tab

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on aa76e5b08aebdd61ebb5362ddcef53c1e7083810<br /><br />RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/20890">issues.joomla.org/tracker/joomla-cms/20890</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/20890">issues.joomla.org/tracker/joomla-cms/20890</a>.</sub>

**Comment by brianteeman** [comment]
Thanks for merging. I will add the same for other components now.

**Comment by helpwise** [comment]
Thanks for this effort Brian et al,..  to make this thing *really* sing, is it possible to add 'Category Template Layout' selector to the new category menu item ?

The Category Menu already has 'template selector', but it only works on single articles and not the category list/blog files (because the list/blog is inherently the category template via an XML file ).
https://github.com/joomla/joomla-cms/issues/19613

Perhaps create a new menu item called "Category (templateable)" so all the old stuff is compatible, and then allow it to properly override both the category and article layouts.. ?

**Comment by ghost** [comment]
@helpwise can you please open a new Issue (and Link to this PR) as Comments on closed Issues didn't get much Notice?

**Comment by helpwise** [comment]
Oh - yes - sorry! :)

**Pull request merged**